src/general/base_stage/CreatureStageHUDBase.cs

```diff
@@ -817,11 +800,7 @@ protected void UpdateReproductionProgress()
             GD.PrintErr("can't get reproduction phosphates progress: ", e);
         }
 
-        ammoniaReproductionBar.Value = fractionOfAmmonia * ammoniaReproductionBar.MaxValue;
-        phosphateReproductionBar.Value = fractionOfPhosphates * phosphateReproductionBar.MaxValue;
-
-        CheckAmmoniaProgressHighlight(fractionOfAmmonia);
-        CheckPhosphateProgressHighlight(fractionOfPhosphates);
+        editorButton.UpdateReproductionProgressBars(fractionOfAmmonia, fractionOfPhosphates, AmmoniaBW, PhosphatesBW);
```

Comment:
I think these black and white textures should be managed by the editor button scene, as that's pretty much a functionality entirely managed by it. I don't think the complexity of needing to pass in specific texture variants to this methods is worth it (as I can't see a reason why the editor button would need to act differently in some special condition).